css-shared-element-transitions/index.bs

```diff
@@ -324,103 +320,124 @@ A {{Document}} additionally has:
 
     : <dfn>ready promise</dfn>
     :: a {{Promise}}.
         Initially [=a new promise=] in [=this's=] [=relevant Realm=].
 
+    : <dfn>DOM updated promise</dfn>
+    :: a {{Promise}}.
+        Initially [=a new promise=] in [=this's=] [=relevant Realm=].
+
     : <dfn>finished promise</dfn>
     :: a {{Promise}}.
         Initially [=a new promise=] in [=this's=] [=relevant Realm=].
 </dl>
 
-The {{SameDocumentTransition/finished}} [=getter steps=] are to return [=this's=] [=SameDocumentTransition/finished promise=].
-
-### {{SameDocumentTransition/prepare()}} ### {#SameDocumentTransition-prepare}
-
-<div algorithm="SameDocumentTransition.prepare()">
-    The [=method steps=] for
-    <dfn method for=SameDocumentTransition>prepare(|cb|)</dfn>
-    are as follows:
-
-    1. If [=this's=] [=SameDocumentTransition/phase=] is not "`idle`", then throw an "{{InvalidStateError}}" {{DOMException}}.
-
-    1. Set [=this's=] [=SameDocumentTransition/phase=] to "`preparing`".
-
-    1. Set [=this's=] [=SameDocumentTransition/prepare callback=] to |cb|.
+The {{DOMTransition/finished}} [=getter steps=] are to return [=this's=] [=DOMTransition/finished promise=].
 
-    1. Let |document| be [=this's=] [=relevant global object's=] [=associated document=].
-
-    1. Let |transitionsToAbandon| be a [=/list=] of every {{SameDocumentTransition}}
-        within [=this's=] [=relevant Realm=]
-        that has a [=SameDocumentTransition/phase=] that is not "`idle`".
-
-        Issue: The ordering needs to be defined, probably in a queue.
+The {{DOMTransition/ready}} [=getter steps=] are to return [=this's=] [=DOMTransition/ready promise=].
 
-    1. [=list/For each=] |transition| of |transitionsToAbandon|,
-        [=abandon the page transition=] |transition| with an "{{AbortError}}" {{DOMException}}.
-
-        Issue: The callbacks may still be running after this, which may not be what we want.
-
-    1. [=Assert=]: |document|'s [=document/pending same document transition outgoing capture=] is null.
+The {{DOMTransition/domUpdated}} [=getter steps=] are to return [=this's=] [=DOMTransition/DOM updated promise=].
 
-    1. Set |document|'s [=document/pending same document transition outgoing capture=] to [=this=].
-
-    1. Return [=this's=] [=SameDocumentTransition/ready promise=].
-
-    Note: This process continues in [=perform an outgoing capture=].
-</div>
-
-### {{SameDocumentTransition/abandon()}} ### {#SameDocumentTransition-abandon}
+### {{DOMTransition/skipTransition()}} ### {#DOMTransition-skipTransition}
 
 <div algorithm>
-    The [=method steps=] for
-    <dfn method for=SameDocumentTransition>abandon()</dfn> are:
+    The [=method steps=] for <dfn method for="DOMTransition">skipTransition()</dfn> are:
 
-    1. If [=this's=] [=SameDocumentTransition/phase=] is not "`finished`",
-        then [=abandon the page transition=] [=this=]
+    1. If [=this=] has not [=DOMTransition/concluded=],
+        then [=skip the page transition=] for [=this=]
```

Comment:
Maybe we add a param to allow skipping the transition without running the callback if that ends up as a common use-case.

Reply:
If we stick with the naming here, I think it'd need to be a different method. Like `abandon` 😄